### DISCUSSION TITLE
Extending the rules engine to other platforms

### DISCUSSION THREAD

**Discussion by rfennell** [tfsaggregator/tfsaggregator#372] [Q&A] [answered]
I have a potential project where there is a need to perform two-way synchronisation of Azure DevOps WI with ServiceNow tickets. Has extending TFSAggregator been considered an option for this type of workload, i.e., adding a ServiceNow API plug-in to the rules engine?

As far as I can see it my other options are:
- A completely bespoke solution, that I think would end up looking a lot like this tool, but with a hard coded rules engine
- Using [Zapier](https://zapier.com) or similar.

**Comment by jessehouwing** [comment, marked as answer]
Certainly never been in the plans for the server plugin.

Which is set for deprecation next year.

But maybe an option for the aggregator-cli?

**Reply by rfennell** [reply]
oops wrong repo I meant to ask about aggregator-cli

**Reply by jessehouwing** [reply]
I'm guessing the wit-sync-migrator would be an even better target...

https://twitter.com/MrHinsh/status/1457689365300326400?t=2fFna5w4KAkjs4E6veJJQQ&s=19

**Reply by rfennell** [reply]
Interesting, I had not thought of going that way. I tend to think of that tool for one-off migration, I forget it can be used for ongoing sync.
